src/document-click.js

```diff
@@ -208,3 +259,61 @@ export function onDocumentElementClick_(e, viewport, history, isIosSafari) {
     win.location.replace(`${curLoc.hash || '#'}`);
   });
 };
+
+/**
+ * Get offset location of click from event taking into account shadowRoot.
+ * @param {!Event} e
+ * @return {!{left: string, top: string}}
+ */
+function getClickLocation_(e) {
+  // Use existence of event path as indicator that event was rewritten
```

Comment:
See previous note about Shadow DOM event target rewriting.  If we have a event.path then we are within a rewrite in which case the event.target matches the host element from which we want to offset.